### PR TITLE
61 setup simple auth0 authentification

### PR DESCRIPTION
**Pull request by deonclem**
Go to `/auth/login` to access the login form. You should be able to log out from a profile once logged in.

image

image

I still need to investigate why it shows the `auth0` domain on the Google login form:

image

The Auth0 application: https://manage.auth0.com/dashboard/eu/dev-cajmwjmi/

The Google OAuth app: https://console.developers.google.com/apis/dashboard?project=tottem

You should have access to both.

It should work fine on production (tottem.app domain) but I have no idea how it will behave / how to properly support dynamic subdomains from the *Now* deployments. The easiest way to test for now is locally.

### CODE REVIEW

**Comment by deonclem** [comment]
Also, clearly not a fan of how the redirection are handled right now but I guess it will be better when we have a proper login button.

File: src/components/Views/Header.tsx [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
+    accent100,
+    accent900,
+    brand100,
+    brand900,
+} from '../../constants/colors'
 import { Logo, Beta } from '../Logo'
 
 const ButtonCTA = styled.a`
-    color: ${brand900};
+    color: ${props => (props.color === 'secondary' ? accent900 : brand900)};
```
Comment (poulainv) [inline src/components/Views/Header.tsx]: Oops, I did not see it, I've just merged something related ...

File: src/lib/auth0-variables.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const domain =
+    process.env.ENV === 'dev' ? 'http://localhost:3000' : 'https://tottem.app'
+export const AUTH_CONFIG = {
+    domain: 'dev-cajmwjmi.eu.auth0.com',
+    clientId: 'tHd8WdWpAXY7B7oZ1M2MFBDpbIVza8yc',
```
Comment (poulainv) [inline src/lib/auth0-variables.js]: Should we rather use now Secrets to store those variables?

**Comment by poulainv** [comment]
What are you worried about dynamic domain? I don't understand everything but we could merge as it does not impact user experience for now. Maybe we should plan a call while we could read all of this together!

**Comment by poulainv** [comment]
Sorry for conflict :/

File: src/components/Views/Header.tsx (hunk 1 above)
Comment (deonclem) [reply src/components/Views/Header.tsx]: Now using StyledButton 🙂

File: src/lib/auth0-variables.js (hunk 2 above)
Comment (deonclem) [reply src/lib/auth0-variables.js]: Yep, working fine with secrets!

**Comment by deonclem** [comment]
> What are you worried about dynamic domain? I don't understand everything but we could merge as it does not impact user experience for now. Maybe we should plan a call while we could read all of this together!

Similar problem explained here: https://spectrum.chat/zeit/now/now-secrets-and-login-url-redirection~4bc64b46-5052-4ac2-9742-636b34be743f

Right now on `https://quiet-cxckges17.now.sh` you will get redirect on `tottem.app` after login (so it doesn't work). There must be a way to dynamically get the deployed domain in the code but I'm not sure how to make Auth0 support it (as I don't think you can use wildcards as authorized domains - i.e. `quiet-*.now.sh`)

Happy to merge for now 🙂

**Pull request merged**